frontend/src/components/UserAuth.js

```diff
@@ -40,23 +43,51 @@ const UserAuth = ({
         transform: 'translate(-50%, -50%)',
       }}
     >
-      <p className="login-title">{pageTitle}</p>
-      <TextField
-        label="Username"
-        variant="standard"
-        value={username}
-        onChange={(e) => setUsername(e.target.value)}
-        sx={{ marginBottom: '1rem' }}
-        autoFocus
-      />
-      <TextField
-        label="Password"
-        variant="standard"
-        type="password"
-        value={password}
-        onChange={(e) => setPassword(e.target.value)}
-        sx={{ marginBottom: '2rem' }}
-      />
+      <p className="userauth-title">{pageTitle}</p>
+      {!usernameSubmittedEmpty && (
```

Comment:
This part also can be refactored, you don't need to have more than 1 Text Field, and should utilize conditional rendering. How you can fix it with just 1 component would be an example like this:

```js
<TextField
  error={usernameSubmittedEmpty}
  autoFocus
  label="Username"
  helperText={usernameSubmittedEmpty ? 'Username cannot be empty' : ''}
  variant="standard"
  value={username}
  onChange={(e) => setUsername(e.target.value)}
  sx={{ marginBottom: '1rem' }}
/>
```

Reply:
Omg, this is so good and I have learnt something new. I like that the code is so much neater now. I have changed it and also incorporated this way of refactoring in my other files.